tests/unit/src/test/scala/tests/DefinitionSuite.scala

```
@@ -135,13 +148,42 @@ class DefinitionSuite extends DirectoryExpectSuite("definition") {
     }
   }
 
-  def shouldHaveDefinition(symbol: String): Boolean = {
-    !symbol.isPackage &&
-    !symbol.startsWith("scala/Any#") &&
-    !symbol.startsWith("scala/Nothing#") &&
-    !symbol.startsWith("scala/Null#") &&
-    !symbol.startsWith("scala/Singleton#") &&
-    !symbol.startsWith("scala/AnyRef#") &&
-    !symbol.startsWith("java/lang/Object#")
-  }
+  private val symbolsWithoutDefition: List[String] =
+    //format: off
+    List(
+      "scala/Any#",
+      "scala/Nothing#",
+      "scala/Null#",
+      "scala/Singleton#",
+      "scala/AnyRef#",
+      "java/lang/Object#",
+      "scala/`|`#",
+      "scala/`&`#"
+    )
+    //format: on
+
+  def shouldHaveDefinition(symbol: String): Boolean =
+    !symbol.isPackage && !symbolsWithoutDefition.exists(v =>
+      symbol.startsWith(v)
+    )
 }
+
+class DefinitionScala2Suite
+    extends DefinitionSuiteBase(
+      inputProps = InputProperties.scala2(),
+      directory = "definition",
+      dialect = dialects.Scala213,
+      badFileNames = List(
+        "ForComprehensions.scala" // local symbols in large for comprehensions cause problems
+      )
+    )
+
+class DefinitionScala3Suite
+    extends DefinitionSuiteBase(
+      inputProps = InputProperties.scala3(),
+      directory = "definition-scala3",
+      dialect = dialects.Scala3,
+      badFileNames = List(
+        "Extension.scala"
```

why is this excluded?

---

To suppress errors because #2627